spec/models/cart_product_spec.rb

```diff
@@ -0,0 +1,132 @@
+RSpec.describe CartProduct, type: :model do
+  describe 'database' do
+    it 'has columns' do
+      expect(subject).to have_db_column(:quantity).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:total_amount).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:cart_id).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:product_id).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:created_at).of_type(:datetime).with_options(null: false)
+      expect(subject).to have_db_column(:updated_at).of_type(:datetime).with_options(null: false)
+    end
+
+    it 'has indexes' do
+      expect(subject).to have_db_index(:cart_id)
+      expect(subject).to have_db_index(:product_id)
+    end
+  end
+
+  describe 'associations' do
+    it 'has relationships' do
+      expect(subject).to belong_to(:cart)
+      expect(subject).to belong_to(:product)
+    end
+  end
+
+  describe 'validations' do
+    context 'when is an attribute supervision' do
+      it 'validates the quantity attribute' do
+        is_expected.to validate_presence_of(:quantity)
+      end
```

Comment:
no need to give a custom message in these validation cases, the default one is just enough
```suggestion
      it { is_expected.to validate_presence_of(:quantity) }
```

Reply:
I agree with what you mention, but in another code review I was told that it was necessary to generate these descriptions for the `it` blocks. I've updated it as you mention as it's extremely verbose leaving it with a description because it's super redundant.

Reply:
We usually prefer to add a description but in this case, its presence doesn't give so much value (maybe it's the only case that we can omit 😅 )